custom_components/adaptive_lighting/switch.py

```diff
@@ -1815,6 +1846,11 @@ async def _service_interceptor_turn_on_handler(
             adaptive_switch = find_switch_for_lights(self.hass, [entity_id])
         except NoSwitchFoundError:
             # This might be a light that is not managed by this AL instance.
+            _LOGGER.debug(
+                "No (or multiple) adaptive switch(es) found for entity %s,"
+                " skipping adaptation by intercepting service call",
+                entity_id,
+            )
```

Comment:
So in #628 (and earlier) I unfortunately missed the fact that a light can have multiple adaptive switches, but I think this can be fixed.

What if we 
1. get all switches (e.g. `find_switches_for_lights`)
2. ask every switch for its desired adaptation (e.g., call `prepare_adaptation_data` multiple times)
3. merge the datas
4. continue with a single `execute_cancellable_adaptation_calls`?

This way we have a single adaptation call sequence managed by a single object instance, the `TurnOnOffListener`. In this case, it probably makes sense to also move the `execute[_cancellable]_adaptation_calls` functions here, just like `cancel_ongoing_adaptation_calls`.

This might also help fixing the actual issue you are trying to solve in this PR, if we also move the event handlers here. The `AdaptiveSwitch`es remain only to manage their settings and provide adaptation hints. We keep a single `adaptation_tasks` list and don't need helpers like the `which`. Could be quite a rewrite though and isn't necessarily meant for this PR - just food for thought.

Reply:
That might be a good solution, however, perhaps hard to implement given all the complexity that already exists. But if you have the energy for it; a very welcome change 😄 

One should only really ever be able to have max 2 switches controlling a single light. And that is only if one switch controls brightness only and the other only color.

Therefore, I think what I added now if sufficient (at least to have a working solution). When running into the situation where one has 2 switches for a single light, the intercepting call doesn't happen, but the old way still works.